systems/trajectory_optimization/dircon_options.cc

```diff
@@ -6,25 +6,121 @@ namespace dairlib {
 namespace systems {
 namespace trajectory_optimization {
 
-DirconOptions::DirconOptions(int n_constraints) {
-  n_constraints_ = n_constraints;
-  is_constraints_relative_ = vector<bool>(n_constraints_);
-  for (int i=0; i < n_constraints_; i++) {
-    is_constraints_relative_[i] = false;
-  }
+DirconOptions::DirconOptions(int n_kin_constraints) {
+  n_kin_constraints_ = n_kin_constraints;
+  is_constraints_relative_ = vector<bool>(n_kin_constraints_, false);
   start_constraint_type_ = DirconKinConstraintType::kAll;
   end_constraint_type_ = DirconKinConstraintType::kAll;
   force_cost_ = 1.0e-4;
 }
+DirconOptions::DirconOptions(int n_constraints,
+                             drake::multibody::MultibodyPlant<double>* plant)
+    : DirconOptions(n_constraints) {
+  n_v_ = plant->num_velocities();
+  n_x_ = plant->num_positions() + plant->num_velocities();
+}
+DirconOptions::DirconOptions(
+    int n_constraints,
+    drake::multibody::MultibodyPlant<drake::AutoDiffXd>* plant)
+    : DirconOptions(n_constraints) {
+  n_v_ = plant->num_velocities();
+  n_x_ = plant->num_positions() + plant->num_velocities();
+}
+
+void DirconOptions::setDynConstraintScaling(double scale, int row_start,
+                                            int row_end) {
+  DRAKE_DEMAND(row_end < n_x_);
+  addConstraintScaling(&dyn_constraint_scaling_, scale, row_start, row_end);
+}
+void DirconOptions::setKinConstraintScaling(double scale, int row_start,
+                                            int row_end) {
+  DRAKE_DEMAND(row_end < n_kin_constraints_);
+  addConstraintScaling(&kin_constraint_scaling_, scale, row_start, row_end);
+}
+void DirconOptions::setImpConstraintScaling(double scale, int row_start,
+                                            int row_end) {
+  DRAKE_DEMAND(row_end < n_v_);
+  addConstraintScaling(&imp_constraint_scaling_, scale, row_start, row_end);
+}
+void DirconOptions::addConstraintScaling(
```

Comment:
`addConstraintScaling` was extracted from `setDynConstraintScaling`, `setKinConstraintScaling` and `setImpConstraintScaling` to avoid code duplication.

Reply:
Ok. I was originally thinking that you could replace this all with one line (see my comment in `addConstraintScaling`, but the `i` from `start` to `end` does represent some code duplication.